faq.md

```diff
@@ -1,45 +1,634 @@
+        
+        protected override void OnUpdateFrame(FrameEventArgs e)
+        {
+			// This gets called every 1/60 of a second.
+            if (KeyboardState.IsKeyDown(Keys.Escape))
+                Close();
+            
+            base.OnUpdateFrame(e);
+        }
+        
+        protected override void OnRenderFrame(FrameEventArgs e)
+        {
+            // Show that we can use OpenGL: Clear the window to cornflower blue.
+            GL.ClearColor(0.39f, 0.58f, 0.93f, 1.0f);
+            GL.Clear(ClearBufferMask.ColorBufferBit);
+        }
+    }
+}
+```
+
+More sophisticated uses will load textures and shaders and models during the `OnLoad()` handler, clean up during the `OnUnload()` handler, and do a lot more than just wait for the user to press `Escape` to close the window — but the code above is enough to get you started.
+
+(Creating a window in OpenTK 3 requires a similar amount of code, but it looks a bit different from this example.  New projects should use OpenTK 4+.)
+
+### How do I call OpenGL/CL/AL functions?
+
+In your code, you can invoke the OpenGL/CL/AL functions directly using the static classes that OpenTK exposes.  OpenTK provides all of the same functions that the underlying native libraries provide, just with the initial `gl*` or `cl*` or `al*` prefix turned into a static class name.  So, for example:
+
+- In C you’d call `int error = glGetError();`.  In C# you’d write `ErrorCode error = GL.GetError();`.
+- In C you’d call `glBindVertexArray(handle);`.  In C# you’d write `GL.BindVertexArray(handle);`.
+- In C you’d write `glCreateShader(GL_FRAGMENT_SHADER);`.  In C# you’d write `GL.CreateShader(ShaderType.FragmentShader)`.
+
+As you can see, the programming models are nearly identical, by design:  If you know how to use OpenGL/OpenCL/OpenAL, you know how to use OpenTK.
+
+**Arrays and pointers**
+
+Passing arrays from .NET to OpenGL/CL/AL (or to any C library) is a complex topic, but OpenTK has built-in support for it:  See the section below on “[How do I pass pointers and arrays?](#How-do-I-pass-pointers-and-arrays)”.
+
+**Constant symbols**
+
+Note in the third example that the constants change syntax slightly:  They aren’t global `#define` directives as in C, but instead have some minor alterations to make them more .NET-friendly:
+
+- The constants are grouped into strong `enum` types like `ErrorCode` and `ShaderType` — they’re not just a big pile of unrelated values like they are in C.
+- The names are changed from `SHOUT_CASE` to `PascalCase` to match C# style conventions.
+- There are a few special enum values, like `All`, that have been added to support cases in C where you’d have written a fixed constant like `0`.
+
+### Where do I start with OpenGL?
+
+Check out [our tutorial here](https://opentk.net/learn/index.html), which will teach you how to build a basic working program that can draw simple objects.  You can learn more about OpenGL in depth over at [Learn OpenGL](https://learnopengl.com/).
```

Comment:
I also think that we should explicitly say that our tutorials are based on https://learnopengl.com

Reply:
Easy fix:

> You can learn more about OpenGL in depth over at [Learn OpenGL](https://learnopengl.com/); our tutorials are based on theirs.